xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import io.grpc.Status;
+
+/**
+ * An {@link XdsClient} instance encapsulates all of the logic for communicating with the xDS
+ * server. It may create multiple RPC streams (or a single ADS stream) for a series of xDS
+ * protocols (e.g., LDS, RDS, VHDS, CDS and EDS) over a single channel. Watch-based interfaces
+ * are provided for each set of data needed by gRPC.
+ *
+ * <p>This class should only be instantiated by the xDS resolver but can be passed to load
+ * balancing policies.
+ */
+abstract class XdsClient {
+
+  /**
+   * Data class containing the results of performing a series of resource discovery RPCs via
+   * LDS/RDS/VHDS protocols. The results may include configurations for path/host rewriting,
+   * traffic mirroring, retry or hedging, default timeouts and load balancing policy that will
+   * be used to generate a service config.
+   */
+  // TODO(chengyuanzhang): content TBD, most information comes from VirtualHost proto.
+  static final class ConfigUpdate {
+    private String clusterName;
+
+    String getClusterName() {
+      return clusterName;
+    }
+
+    static Builder newBuilder() {
+      return new Builder();
+    }
+
+    static final class Builder {
+      private ConfigUpdate base;
+
+      private Builder() {
```

Comment:
The constructor is not accessible outside the abstract class.

Reply:
You just call `ConfigUpdate.newBuilder()`.

Reply:
I see.